packages/nft/src/NFTHandler.ts

```diff
@@ -240,4 +235,70 @@ export class NFTHandler {
     }
     return tokenUri
   }
+
+  /**
+   * It takes an address and returns his balance
+   * @param {NFT721Manager} nftManager - NFT721Manager - The NFT721Manager instance that will be used
+   * to fetch the balance.
+   * @param {string} address - The address of which we want the balance.
+   * @returns The balance
+   */
+  @ManagerMustActive
+  private static async getAddressBalance(
+    nftManager: NFT721Manager,
+    address: string
+  ): Promise<number> {
+    if (!nftManager.implements('balanceOf', ['address'])) {
+      throw FUNCTION_NOT_IMPLEMENTED(
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        nftManager.contractAddr!,
+        'balanceOf(address)'
+      )
+    }
+
+    try {
+      const { value } = await nftManager.execute({
+        method: 'balanceOf(address)',
+        params: [address]
+      })
+      return value
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      throw BALANCE_OF_FAILED(address, error)
+    }
+  }
+
+  /**
+   * Takes an address and an index and returns the corresponding tokenUri.
+   * @param {NFT721Manager} nftManager - NFT721Manager - The NFT721Manager instance that will be used
+   * to fetch the tokenUri.
+   * @param {string} owner - The owner of the nft.
+   * @param {number} index - The index of the tokenUri.
+   * @returns The balance
+   */
+  @ManagerMustActive
+  private static async tokenOfOwnerByIndex(
+    nftManager: NFT721Manager,
+    owner: string,
+    index: number
+  ): Promise<string> {
+    if (!nftManager.implements('tokenOfOwnerByIndex', ['address', 'uint256'])) {
+      throw FUNCTION_NOT_IMPLEMENTED(
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        nftManager.contractAddr!,
+        'tokenOfOwnerByIndex(address,uint256)'
+      )
+    }
+
+    try {
+      const { value } = await nftManager.execute({
+        method: 'tokenOfOwnerByIndex(address,uint256)',
+        params: [owner, index]
+      })
+      return value
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      throw TRANSACTION_FAILED(error)
```

Comment:
I think that the execute already throws an error if the execute fails, is it necessary to try/catch it?